src/http2_stream.erl

```diff
@@ -574,10 +580,17 @@ check_content_length(Stream) ->
         undefined ->
             ok;
         _Other ->
-            case Stream#stream_state.request_body_size =:= binary_to_integer(ContentLength) of
-                true ->
-                    ok;
-                false ->
+            try binary_to_integer(ContentLength) of
+                Integer ->
+                    case Stream#stream_state.request_body_size =:= Integer of
+                        true ->
+                            ok;
+                        false ->
+                            rst_stream_(?PROTOCOL_ERROR, Stream),
+                            rst_stream
+                    end
+            catch
+                _:_ ->
```

Comment:
in `try <exp1> of <pat> -> <exp2> catch <pat2> -> <exp3>`, only `<exp1>` is protected. 

Still, I like to be as strict as possible, so catching `error:badarg` sounds good there since:

```erlang
2> try binary_to_integer(<<"lkfjd">>) of N -> N catch error:badarg -> caught end.
caught
```
This will still allow failures for cases, for example, where an old version of Erlang was used to compile and `error:undef` will still crash the code as expected.